pkg/controller/loadtest.go

```diff
@@ -275,7 +275,7 @@ func (c *Controller) processNextWorkItem() bool {
 // converge the two. It then updates the Status block of the LoadTest resource
 // with the current status of the resource.
 func (c *Controller) syncHandler(key string) error {
-	ctx, cancel := context.WithTimeout(context.Background(), kubekangal.KubeTimeout)
+	ctx, cancel := context.WithTimeout(context.Background(), syncHandlerTimeout)
```

Comment:
Hard coding this could cause a similar issue that you are trying to fix with `KUBE_CLIENT_TIMEOUT` maybe add a to do to introduce a `KUBE_WAIT_FOR_TIMEOUT` or something?

Also maybe the split of `kubekangal.KubeTimeout` should be split into 2 PR's?

Reply:
Partially explained in https://github.com/hellofresh/kangal/pull/106#discussion_r562691500; the entire `syncHandler` should not be wrapped with a 30 seconds "kube timeout". It was a somewhat lazy way to prevent kube timeout, which worked until the entire sync process took longer than 30 seconds.  
This is now done by adding 5 seconds timeout to each individual operation done by `kubeClient`.

I previously considered removing `syncHandlerTimeout` since the individual timeouts should be sufficient, but I added a 60 seconds timeout in case it does get too long. Now I think the better approach is perhaps to add a metric to track the time taken for the sync process (and have alerts if it's taking too long).

Reply:
Looking at this function and the amount of calls it is doing to `cache`, `kubernetes`, `backend` I think making this configurable or at least based on something configurable maybe best.
For example let's say you set kubeClientTimeout to 2 minutes then this sync will now always start failing assuming k8s responds after 60s. 

It is probably best to allow a user to specify this timeout or to calculate it based on timeouts set for k8s, backend, cache etc.

Reply:
It is now an environment variable (fd31a83)